### PR TITLE
Fix an issue with deleting frames

### PR DESCRIPTION
**Pull request by klakhov**
### Motivation and context

Previously, in the Honeypots patch, we introduced periodic metadata updates to allow the UI to reflect chunk changes dynamically. However, this introduced a couple of issues:  

1. The object received from the server was incorrectly utilized by the UI, preventing frame deletion after requesting updated metadata.  
2. The new object overwritten unsaved deleted frame data, leading to the loss of user data.  

### Steps to Reproduce the Frame Deletion Issue  
1. Set `jobMetaDataReloadPeriod` in `config.ts` of `cvat-core` to a small interval, such as 15 seconds (the default is 1 hour).  
2. Open a job and ensure the `/data/meta` request is sent.  
3. Wait for the `jobMetaDataReloadPeriod` to elapse.  
4. Change the frame and confirm that a new `/data/meta` request is sent.  
5. Attempt to delete any frame and observe that it is not deleted.  
deleting-frame

### Resolution  
To fix this, I removed the secondary storage of `jobMeta` in `frameDataCache`. Now, it uses an asynchronous getter that fetches data directly from `frameMetaCache`. Having two caches for the same data caused synchronization issues, which led to this bug.  

To fix second issue I added merge logic inside getFramesMeta function that will update new object with unsaved data.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- ~~[ ] I have updated the documentation accordingly~~
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new enum to standardize the representation of deleted frame states.
	- Enhanced metadata retrieval efficiency with updated caching logic.

- **Bug Fixes**
	- Improved clarity and maintainability in handling the deletion status of frames.

- **Refactor**
	- Simplified logic for marking frames as deleted or restored using the new enum.

### CODE REVIEW

File: cvat-core/src/frames.ts [hunk 1]
```diff
@@ -555,7 +564,27 @@ export function getFramesMeta(type: 'job' | 'task', id: number, forceReload = fa
                     ...serialized,
                     deleted_frames: Object.fromEntries(serialized.deleted_frames.map((_frame) => [_frame, true])),
                 });
-                resolve(framesMetaData);
+                // When we get new framesMetaData from server there can be some unsaved data
+                // here we merge new meta data with cached one
```
Comment (zhiltsov-max) [inline cvat-core/src/frames.ts:568]: Consider extracting a function for this.

File: cvat-core/src/frames.ts [hunk 2]
```diff
@@ -555,7 +564,27 @@ export function getFramesMeta(type: 'job' | 'task', id: number, forceReload = fa
...
+                if (previousCache instanceof Promise) {
+                    previousCache.then((prevMeta) => {
+                        const updatedFields = prevMeta.getUpdated();
+                        for (const key in updatedFields) {
+                            if (Object.hasOwn(updatedFields, key) && key.startsWith('deletedFrames')) {
```
Comment (zhiltsov-max) [inline cvat-core/src/frames.ts:573]: Why not `key === 'deletedFrames'`?

File: cvat-core/src/frames.ts (hunk 1 above)
Comment (klakhov) [reply cvat-core/src/frames.ts:605]: Moved to separate function

File: cvat-core/src/frames.ts (hunk 2 above)
Comment (klakhov) [reply cvat-core/src/frames.ts]: Makes sense. Moved all logic with keys into FramesMetaData class, now we will just use `.deletedFrames` prop from `.getUpdated()`

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**